lib/convict.js

```diff
@@ -171,6 +171,16 @@ function validate(instance, schema, strictValidation) {
     }
     delete flatInstance[name];
 
+    // ignore nested keys of 'object' config params 
+    if (schemaItem.format === 'object') {
+      Object.keys(flatInstance)
+        .filter(function(key) {
+          return key.startsWith(name + '.')
```

Comment:
To avoid needing to change the version of Node and have an associated major version bump, go ahead and replace this with `return key.lastIndexOf(name + '.', 0) === 0;`